### PR TITLE
Jest matchers can accept many arguments

### PR DESCRIPTION
**Pull request by JamieMason**
This PR removes the restriction on the number of arguments Jest's custom matchers can accept, which does not exist in Jest itself.

Jest allows you to define the following custom matcher for example which accepts 2 arguments:

```js
expect.extend({
  toBeWithinRange(received, floor, ceiling) {
    // implementation...
  }
});
```

The Jest typings currently state that a Jest matcher can only accept one argument.

A URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/jest/docs/en/expect.html#expectextendmatchers.

Thanks.

### CODE REVIEW

**Comment by wsmd** [comment]
Hi @JamieMason, great catch! Thanks for a quick fix. I'm happy to approve this PR, but do you mind adding this use case to [`jest-tests.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jest/jest-tests.ts#L253-L278)?

**Review by wsmd — CHANGES_REQUESTED** [review]
Test needed

**Comment by JamieMason** [comment]
will do! Thanks

**Comment by RyanCavanaugh** [comment]
@JamieMason Please address comments from the code reviewers.

**Comment by JamieMason** [comment]
Hi all, I have updated this PR to include a new test case. Please let me know if there is anything more I can do, thanks a lot.

**Comment by RyanCavanaugh** [comment]
@wsmd - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by wsmd — APPROVED** [review]
Looks good! - Thanks for the fix 👍

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
